### PR TITLE
 Read registry params from env file

### PR DESCRIPTION
**Pull request by pawanpinjarkar**

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -87,6 +87,15 @@ const ignitionConfigFormat = `{
...
   },
+  "storage": {
+	  "files": [{
+		  "filesystem": "root",
+		  "path": "/etc/containers/registries.conf",
+		  "mode": 420,
+		  "overwrite": true,
+		  "contents": { "source": "data:text/plain;charset=utf-8,%5Bregistries.insecure%5D%0Aregistries%3D%5B%22local.registry%3A5001%22%5D" }
+	  }]
+  },
```
Comment (sacharya) [inline internal/bminventory/inventory.go]: Did  you mean to change this ?

File: onprem-environment [hunk 2]
```diff
@@ -15,4 +15,8 @@ AGENT_DOCKER_IMAGE=local.registry:5001/ocpmetal/agent:latest
 CONNECTIVITY_CHECK_IMAGE=local.registry:5001/ocpmetal/connectivity_check:latest
 INVENTORY_IMAGE=local.registry:5001/ocpmetal/inventory:latest
 HARDWARE_INFO_IMAGE=local.registry:5001/ocpmetal/hardware_info:latest
-FREE_ADDRESSES_IMAGE=local.registry:5001/ocpmetal/free_addresses:latest
\ No newline at end of file
+FREE_ADDRESSES_IMAGE=local.registry:5001/ocpmetal/free_addresses:latest
+MIRROR_REPO=local.registry:5000/ocp4
```
Comment (sacharya) [inline onprem-environment]: Call it IMAGE_CONTENT_MIRROR

File: onprem-environment [hunk 3]
```diff
@@ -15,4 +15,8 @@ AGENT_DOCKER_IMAGE=local.registry:5001/ocpmetal/agent:latest
...
+MIRROR_SOURCE_IMAGE_1=quay.io/openshift-release-dev/ocp-release
```
Comment (sacharya) [inline onprem-environment]: Call it IMAGE_CONTENT_SOURCE_1

File: onprem-environment (hunk 2 above)
Comment (pawanpinjarkar) [reply onprem-environment]: Done

File: onprem-environment (hunk 3 above)
Comment (pawanpinjarkar) [reply onprem-environment]: Done

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (pawanpinjarkar) [reply internal/bminventory/inventory.go]: No, fixed it

File: internal/installcfg/installcfg.go [hunk 4]
```diff
@@ -136,23 +138,23 @@ func getBasicInstallConfig(cluster *common.Cluster) *InstallerConfigBaremetal {
...
 		ImageContentSources: []struct {
-			Mirrors []string `yaml:"mirrors"`
-			Source  string   `yaml:"source"`
+			Mirrors []string `yaml:"mirrors", envconfig:"MIRROR_REPO"`
+			Source  string   `yaml:"source", envconfig:"MIRROR_SOURCE_IMAGE_1", envconfig:"MIRROR_SOURCE_IMAGE_2"`
 		}{
 			{
 				Mirrors: []string{
-					"local.registry:5000/ocp4",
+					os.Getenv("MIRROR_REPO"),
```
Comment (sacharya) [inline internal/installcfg/installcfg.go]: Not a good idea to read from env randomly in the middle of the code. How are rest of the variables read from the file?

File: onprem-environment [hunk 5]
```diff
@@ -15,4 +15,8 @@ AGENT_DOCKER_IMAGE=local.registry:5001/ocpmetal/agent:latest
 CONNECTIVITY_CHECK_IMAGE=local.registry:5001/ocpmetal/connectivity_check:latest
 INVENTORY_IMAGE=local.registry:5001/ocpmetal/inventory:latest
 HARDWARE_INFO_IMAGE=local.registry:5001/ocpmetal/hardware_info:latest
-FREE_ADDRESSES_IMAGE=local.registry:5001/ocpmetal/free_addresses:latest
\ No newline at end of file
+FREE_ADDRESSES_IMAGE=local.registry:5001/ocpmetal/free_addresses:latest
+IMAGE_CONTENT_MIRROR=local.registry:5000/ocp4
+IMAGE_CONTENT_SOURCE_1=quay.io/openshift-release-dev/ocp-release
+MIRROR_SOURCE_IMAGE_2=quay.io/openshift-release-dev/ocp-v4.0-art-dev
+ADDITIONAL_TRUST_BUNDLE=<YOUR_REGISTRY_CA_CERT_HERE>
```
Comment (sacharya) [inline onprem-environment]: What format should this be provided in?

File: internal/installcfg/installcfg.go (hunk 4 above)
Comment (pawanpinjarkar) [reply internal/installcfg/installcfg.go]: Rest ones are read as struct.field but I am not sure how will it work when I change `Source: os.Getenv("IMAGE_CONTENT_SOURCE_2"),` to `Source: InstallerConfigBaremetal.Source,` I will test it on a working machine.

Comment (pawanpinjarkar) [reply internal/installcfg/installcfg.go]: @sacharya The dependency `"github.com/kelseyhightower/envconfig"` currently has no support for a slice of structs but they do have an open PR here https://github.com/kelseyhightower/envconfig/pull/170. Till then should we continue to read from env variables? `AdditionalTrustBundle` can be changed not to read from env var but `ImageContentSources` will still need to be. WDYT?

File: onprem-environment (hunk 5 above)
Comment (pawanpinjarkar) [reply onprem-environment]: How about this? `YOUR_REGISTRY_CA_CERT_MINIFIED_STRING_HERE`

Comment (sacharya) [reply onprem-environment]: Have you tried testing it in this format? Do we need to minify here? 
ADDITIONAL_TRUST_BUNDLE="BEGIN CERTIFICATE-----
    .....................
    END CERTIFICATE-----"

File: internal/installcfg/installcfg.go (hunk 4 above)
Comment (pawanpinjarkar) [reply internal/installcfg/installcfg.go]: With reference to the above comment, I am proceeding with reading from env var.

File: onprem-environment (hunk 5 above)
Comment (pawanpinjarkar) [reply onprem-environment]: The trust bundle ca-cert needs to be a minified string else podman fails to start the installer container as the `--env-file` flag can't parse multiline value trust bundle cert.

**Comment by pawanpinjarkar** [comment]
Ref https://github.com/sacharya/bm-inventory/pull/2

**Pull request closed without merging**